prog/dftb+/lib_dftbplus/mainio.F90

```diff
@@ -2899,12 +2957,17 @@ subroutine writeDetailedOut1(fd, iDistribFn, nGeoSteps, iGeoStep, tMD, tDerivs,
           & energy%EForceRelated * Hartree__eV, 'eV'
     end if
     if (tPeriodic .and. pressure /= 0.0_dp) then
-      write(fd, format2U) 'Gibbs free energy', energy%Etotal - sum(TS) + cellVol * pressure,&
-          & 'H', Hartree__eV * (energy%Etotal - sum(TS) + cellVol * pressure), 'eV'
+      write(fd, format2U) 'Gibbs free energy', energy%EGibbs, 'H', Hartree__eV * energy%EGibbs, 'eV'
     end if
     write(fd, *)
 
     if (tAtomicEnergy) then
+
+      ! MBD does not calculate any atomic contributions
+      if (tManyBodyDisp) then
+        call error("Many-body dispersion not compatible with atomic energy printing")
+      end if
```

Comment:
This error should come up in `initprogram` not in every SCC step, if it all. Probably it straight-forward to just distribute the dispersion energy by `eMbd / real(nAtom, dp)` and have some dummy atom-resolved energy like it is done with the DFT-D3 integration.